spec/process_secret_spec.js

```diff
@@ -47,7 +47,7 @@ describe('Process Secret', function() {
       ]
     }
     await processSecrets(secretValue, secretConfig);
-    const testFileContents = fs.readFile(TEST_FILE_SECRET, 'utf8');
+    const testFileContents = await fs.readFile(TEST_FILE_SECRET, 'utf8');
```

Comment:
for all async tests we should call `done` method https://jasmine.github.io/tutorials/async to make sure our tests don't fail randomly due to race conditions in code